openquake/hazardlib/gsim/cauzzi_2014.py

```diff
@@ -0,0 +1,513 @@
+    spectra and peak ground motions, based on a global dataset of digital
+    acceleration records", Bulletin of Earthquake Engineering, In Press
+
+    Spectral acceleration (SA) values are obtained from displacement response
+    spectrum  (DSR) values (as provided by the original equations) using the
+    following formula ::
+
+        SA = DSR * (2 * π / T) ** 2
+
+    """
+    #: Supported tectonic region type is active shallow crust,
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are spectral acceleration, peak
+    #: ground acceleration and peak ground velocity.
+    #: The original paper provides coefficients for PGA and PGV, while SA
+    #: is obtained from displacement response spectrum values.
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV,
+        SA
+    ])
+
+    #: Supported intensity measure component is the geometric mean of two
+    #: horizontal components
+    #: :attr:`~openquake.hazardlib.const.IMC.AVERAGE_HORIZONTAL`,
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation types are inter-event, intra-event and
+    #: total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT,
+        const.StdDev.TOTAL
+    ])
+
+    #: Required site parameter is only Vs30
+    REQUIRES_SITES_PARAMETERS = set(('vs30', ))
+
+    #: Required rupture parameters are magnitude and rake
+    REQUIRES_RUPTURE_PARAMETERS = set(('rake', 'mag'))
+
+    #: Required distance measure is Rrup,
+    REQUIRES_DISTANCES = set(('rrup', ))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # extract dictionaries of coefficients specific to required
+        # intensity measure type
+        C = self.COEFFS[imt]
+
+        mean = self._compute_mean(C, rup, dists, sites, imt)
+
+        stddevs = self._get_stddevs(C, stddev_types, sites.vs30.shape[0])
+
+        return mean, stddevs
+
+    def _compute_mean(self, C, rup, dists, sites, imt):
+        """
+        Returns the mean ground motion acceleration and velocity
+        """
+        mean = (self._get_magnitude_scaling_term(C, rup.mag) +
+                self._get_distance_scaling_term(C, rup.mag, dists.rrup) +
+                self._get_style_of_faulting_term(C, rup.rake) +
+                self._get_site_amplification_term(C, sites.vs30))
+        # convert from cm/s**2 to g for SA and from cm/s**2 to g for PGA (PGV
+        # is already in cm/s) and also convert from base 10 to base e.
+        if isinstance(imt, PGA):
+            #mean = np.log((100 ** mean) / g)
+            mean = np.log((10 ** mean) * ((2 * np.pi / 0.01) ** 2) *
```

Comment:
Here I would add a comment saying that PGA is actually PSA(0.01s;5%) Cauzzi et al., 2014, page XX (I miss pages in my .pdf)